### PR TITLE
EDM-449: hooks: fix oneOf syntax

### PR DESCRIPTION
**Pull request by hexfusion**
This PR fixes a misconfigured oneOf spec. This was a new approach and my previous usage was not working as expected.

### CODE REVIEW

**Comment by hexfusion** [comment]
cc @celdrake

**Comment by celdrake** [comment]
> cc @celdrake

@hexfusion  I can confirm the type generation works with the branch. The `HookAction` type is generated as:
```
export type HookAction = {
  executable?: HookActionExecutableSpec;
  systemd?: HookActionSystemdSpec;
};
```
which seems to match the new definition.

Thanks a lot!

**Review by avishayt — APPROVED** [review]

**Review by avishayt — APPROVED** [review]

File: api/v1alpha1/util.go [hunk 1]
```diff
@@ -12,18 +11,19 @@ const (
 
 // Type returns the type of the action.
 func (t HookAction) Type() (string, error) {
-	var data map[string]interface{}
-	if err := json.Unmarshal(t.union, &data); err != nil {
-		return "", err
+	// The OpenAPI spec will prevent this from happening, but we should still
+	// evaluate for go code usage.
```
Comment (avishayt) [inline api/v1alpha1/util.go:15]: Are you sure that the OpenAPI spec will prevent this?  I'm not familiar with the syntax you used and don't see the check in the generated code.

Comment (hexfusion) [reply api/v1alpha1/util.go:15]: yeah I am reworking this

**Pull request closed without merging**